scipy/signal/spectral.py

```diff
@@ -826,8 +850,13 @@ def detrend_func(d):
             scale = 1.0 / win.sum()**2
         else:
             raise ValueError('Unknown scaling: %r' % scaling)
-    else:
-        scale = 1
+    elif mode == 'complex':
+        if scaling == 'density':
+            scale = np.sqrt(1.0 / (fs * (win*win).sum()))
+        elif scaling == 'spectrum':
+            scale = 1.0 / win.sum()
+        else:
+            raise ValueError('Unknown scaling: %r' % scaling)
```

Comment:
I'm not sure why we would want to check these factors via computation. Since the PSD has amplitude that is the square of the complex spectrum amplitude, I just used the square root of the PSD window correction factors. 

In any case, this is effectively tested  in `TestSpectrogram.test_cross_spectrogram`, which compares the direct CSD calculation to what you get from two complex spectrograms. 

It would maybe be clearer, though, to change the logical flow here. I.e. just define the scaling factors for `density` vs `scaling`, and then take the square root if `mode=='complex'`.